Lib/pathlib.py

```diff
@@ -198,14 +213,34 @@ def resolve(self, path, strict=False):
         # Means fallback on absolute
         return None
 
-    def _split_extended_path(self, s, ext_prefix=ext_namespace_prefix):
+    def _split_extended_path(self, s, ext_prefix=ext_namespace_prefix,\
+                             ext_prefix2=ext_namespace_prefix2):
+        # See https://bugs.python.org/issue33898
         prefix = ''
-        if s.startswith(ext_prefix):
+        if s.startswith(ext_prefix) or s.startswith(ext_prefix2):
             prefix = s[:4]
             s = s[4:]
-            if s.startswith('UNC\\'):
-                prefix += s[:3]
-                s = '\\' + s[3:]
+            index = s.find('\\')
+            if index != -1:
+                # Do not assume case sensitivity.
+                # See https://docs.microsoft.com/ru-ru/windows/desktop/FileIO/naming-a-file
+                # But .lower() is not recommended. See https://bugs.python.org/issue32612
+                s1 = s[:index]
+                if s1 == 'Global' or s1 == 'GLOBAL':
+                    prefix += s[:6]
+                # For example, Path('//?/Global/Z:/').drive
+                    if s[8] == ':':
+                        prefix += s[6:7]
+                        s = s[7:]
+                # For example, r'\\?\Global\UNC\server\share'
+                    elif s[7:10] == 'UNC':
```

Comment:
Should we check `s[7:11] == 'UNC\\'`? And I think the returned prefix in this case should include the server and share as well, just like when you use the `\\server\share` format (but if you know some reason this is not the case, happy to leave it this way).

Reply:
In the previous version it checked the start of the line if it's 'UNC\\'. 
Now before that line there is index = s.find('\\'). So it checks the part that ends with already found '\\'.
 index = s.find('\\')
            if index != -1:
                s1 = s[:index]
                if s1 == 'UNC':

If find() is not recommended here, it can be returned to the previous form (s1 is used for 'Global' too).

Except for this grammatics and adding '\\.\' prefix, I didn't change the parsing of UNC paths.  _split_extended_path() separates prefix, while server and share are later parsed by splitroot().

Reply:
Fixed the missing trailing slash in 'UNC' coming after 'Global'.